queue_test.go

```diff
@@ -379,11 +380,75 @@ func (suite *QueueSuite) TestConsuming(c *C) {
 	connection := OpenConnection("consume", "tcp", "localhost:6379", 1)
 	queue := connection.OpenQueue("consume-q").(*redisQueue)
 
-	c.Check(queue.StopConsuming(), Equals, false)
+	wg := queue.StopConsuming()
+	c.Check(wg, NotNil)
 
 	queue.StartConsuming(10, time.Millisecond)
-	c.Check(queue.StopConsuming(), Equals, true)
-	c.Check(queue.StopConsuming(), Equals, false)
+	c.Check(queue.StopConsuming(), NotNil)
+}
+
+func (suite *QueueSuite) TestStopConsuming_Consumer(c *C) {
+	connection := OpenConnection("consume", "tcp", "localhost:6379", 1)
+	queue := connection.OpenQueue("consume-q").(*redisQueue)
+	queue.PurgeReady()
+
+	deliveryCount := 30
+
+	for i := 0; i < deliveryCount; i++ {
+		queue.Publish("d" + strconv.Itoa(i))
+	}
+
+	queue.StartConsuming(20, time.Millisecond)
+	var consumers []*TestConsumer
+	for i := 0; i < 10; i++ {
+		consumer := NewTestConsumer("c" + strconv.Itoa(i))
+		consumers = append(consumers, consumer)
+		queue.AddConsumer("consume", consumer)
+	}
+
+	finishedChan := queue.StopConsuming()
+	c.Assert(finishedChan, NotNil)
+
+	<-finishedChan
+
+	var consumedCount int
+	for i := 0; i < 10; i++ {
+		consumedCount += len(consumers[i].LastDeliveries)
+	}
+
+	// make sure all fetched deliveries are consumed
+	c.Check(consumedCount, Equals, deliveryCount-queue.ReadyCount())
+	c.Check(queue.deliveryChan, HasLen, 0)
+
+	connection.StopHeartbeat()
+}
+
+func (suite *QueueSuite) TestStopConsuming_BatchConsumer(c *C) {
+	connection := OpenConnection("batchConsume", "tcp", "localhost:6379", 1)
+	queue := connection.OpenQueue("batchConsume-q").(*redisQueue)
+	queue.PurgeReady()
+
+	deliveryCount := 30
+
+	for i := 0; i < deliveryCount; i++ {
+		queue.Publish("d" + strconv.Itoa(i))
+	}
+
+	queue.StartConsuming(20, time.Millisecond)
+	consumer := NewTestBatchConsumer()
+	consumer.AutoFinish = true
+	queue.AddBatchConsumer("consume", 50, consumer)
```

Comment:
Maybe lets change the numbers here a bit. Currently you publish 30 deliveries for a batch consumer with limit 50 and prefetch limit 20. So it's only going to be a single batch. Maybe change the numbers so it actually creates multiple batches. Maybe deliveryCount=100, batchLimit=10?